### DISCUSSION TITLE
Multiexpression Outputs Question

### DISCUSSION THREAD

**Discussion by Alexia369** [MilesCranmer/PySR#739] [Q&A]
```
function my_loss_function(tree, dataset::Dataset{T,L}, options)::L where {T,L}

  tree.degree != 2 && return L(Inf)

    P = tree.l
    Q = tree.r

    # Evaluate numerator:
    P_prediction, flag = eval_tree_array(P, dataset.X, options)
    !flag && return L(Inf)

    # Evaluate denominator:
    Q_prediction, flag = eval_tree_array(Q, dataset.X, options)
    !flag && return L(Inf)
```
If there are 3 or more multi-expression outputs, how to realize and distinguish them?
You say,Want to make a new thread for that? But basically just apply the pattern recursively.
How to achieve it in a simple way? Is there any code? Thanks

**Comment by MilesCranmer** [comment]
This should be **significantly easier** with the upcoming TemplateExpression. ETA 2 days for SR.jl and TBD for PySR.

**Reply by Alexia369** [reply]
Thank you. If there is progress in the template expression, where can we know?
